### PR TITLE
StyleGAN2 implementation on OASIS Brain dataset

### PR DESCRIPTION
**Pull request by OliverRuffell**
Hi Shakes and the team,

I have implemented the StyleGAN2 for the OASIS Brain dataset to create reasonably clear images of MRI brain scans.

The files included are:
- **ModConv2D.py** -> this is a file that contains the class for a custom keras layer to modulate the weights of the normal Conv2D
- **README.md** -> this explains what was set out to do, shows some results, and explains how to use the files
- **StyleGAN2.ipynb** -> this is the jupyter notebook file that was used to quickly test out things with the networks
- **MNIST test stuff/StyleGAN_MNIST.ipynb** -> this was used fort testing the models on the MNIST dataset and has an example of how to adapt the networks to work with a different dataset
- **driver.py** -> this file has the ability to modify the parameters of the model, and is run to train the model
- **functions.py** -> this was used to hold additional functions such as the training functions, and some graphing, as well as model periphery functions
- **models.py** -> this file was used to make the models
- **images/** -> this folder contains some images used in the README.md The two images are a gif of training images, and an image of the final epoch (after 30 epochs, ~8 hours)

Thanks,

Oliver R.
45320156

### CODE REVIEW

**Review by WeiDaiwow — COMMENTED** [review]
Have both test and driver file
Good commits 
Nice readme file
An algorithm solves the problems
Good design and nice implementation
Nice comments
Nice description and explanation of the algorithm
Correct pull request with the description
nice visualization 
Dependencies included
Nice output and plots
Nice Markdown

Suggestions:
1. please show some plots during the training such as convergence plot or loss plot for both generator and discriminator
2. Please include some references in the readme file

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description good
